src/App/Components/DisplayArticle/readArticle.js

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable max-len */
+/* eslint-disable react/destructuring-assignment */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import propTypes from 'prop-types';
+import Popular from '../Popular/popular';
+import ReadArticleActions from '../../../Redux/Actions/readArticleActions';
+import PopularArticleAction from '../../../Redux/Actions/readPopularActions';
+import ArticleNotFound from '../ArticleNotFound/ArticleNotFound';
+import BreadCrumb from './breadCrumb';
+import ArticleBody from './articleBody';
+import Loader from '../Loader/Loader';
+
+const { readArticle, getTags } = ReadArticleActions;
+const { readPopularArticle } = PopularArticleAction;
+
+export class ReadArticle extends Component {
+  state = { tags: [], isLoading: true, articles: [{ title: '', User: {}, Category: {} }] }
+
+  componentWillMount() {
+    const { readArticle: readArticles, getTags: getTag, readPopularArticle: readPopular } = this.props;
+    const { match: { params: { slug } } } = this.props;
+    readArticles(slug);
+    getTag(slug);
+    readPopular();
+  }
+
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.article) {
+      this.setState({
+        slug: nextProps.article.slug,
+        category: nextProps.article.Category.name,
+        title: nextProps.article.title,
+        body: nextProps.article.body,
+        readTime: nextProps.article.readTime,
+        createdAt: nextProps.article.createdAt,
+        coverImage: nextProps.article.coverImage,
+        firstName: nextProps.article.User.firstName,
+        lastName: nextProps.article.User.lastName,
+        profileImage: nextProps.article.User.profileImage
+      });
+    }
+
+    setTimeout(() => {
```

Comment:
@MCFrank16 why do you need this `setTimeout`?

Reply:
I am using this to set a delay of 2 seconds loading properly data from the database